src/OpenTelemetry.Api/Context/Propagation/CompositePropagator.cs

```diff
@@ -0,0 +1,86 @@
+﻿// <copyright file="CompositePropagator.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+namespace OpenTelemetry.Context.Propagation
+{
+    /// <summary>
+    /// CompositePropagator provides a mechanism for combining multiple propagators into a single one.
+    /// </summary>
+    public class CompositePropagator : ITextFormat
+    {
+        private readonly List<ITextFormat> textFormats;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CompositePropagator"/> class.
+        /// </summary>
+        /// <param name="textFormats">List of <see cref="ITextFormat"/> wire context propagator.</param>
+        public CompositePropagator(List<ITextFormat> textFormats = null)
+        {
+            this.textFormats = textFormats ?? new List<ITextFormat>();
```

Comment:
what is the use case for having empty list?

Reply:
The idea of this is: if we receive a null list, the code would act the same as if we receive a list with at least one element.
Question:
- should we maintain this?
- should we change to `if list null or empty throw exception`?

Reply:
If the list is empty, I think we should throw. - Reiley made suggestion about the overall principles  - its okay to throw at initialization stage if bad input is received. But we shouldn't ever throw after startup.

List with '0' element - no need of throwing. But no injection/extraction occurs.

Reply:
just updated!